netty/src/main/java/io/grpc/netty/NettyClientTransport.java

```diff
@@ -194,6 +199,27 @@ public ClientStream newStream(
     if (channel == null) {
       return new FailingClientStream(statusExplainingWhyTheChannelIsNull, tracers);
     }
+    try {
+      if (callOptions.getAuthority() != null && !negotiator.mayBeVerifyAuthority(
+          callOptions.getAuthority())) {
+        String errMsg = String.format("Peer hostname verification failed for authority '%s'.",
+            callOptions.getAuthority());
+        logger.log(Level.FINE, errMsg);
+        return new FailingClientStream(Status.INTERNAL.withDescription(errMsg), tracers);
```

Comment:
`UNAVAILABLE` would be better here. `INTERNAL` is when we detect a bug in gRPC and can't communicate or such. TLS failures normally show up as `UNAVAILABLE`.

Reply:
Done.